scipy/stats/_qmc.py

```diff
@@ -1081,7 +1079,7 @@ def integers(
 
         return sample
 
-    def reset(self) -> QMCEngine:
+    def reset(self) -> "QMCEngine":
```

Comment:
@jorenham I see in scipy-stubs you use `Self` from `typing_extensions` for this. Is this an acceptable way to do things on Python 3.10 without `typing_extensions`?

Reply:
`Self` is indeed the best option for cases like these, if it's available. If not, then there are several alternatives:

1. use `.pyi` stubs instead, and `from typing_extensions import Self`, which always is allowed, even if `typing_extensions` is not available (because `typeshed` bundles the `typing_extensions` stubs as part of the stdlib)
2. wrap the method in `sys.version_info >= (3, 10)` and add `from typing import Self` within there. See 3. for what to do within the `else:`.
3. declare `_SelfT = TypeVar("_SelfT")`, and annotate `def reset(self: _SelfT) -> _SelfT: ...` (this is the most common approach, and comes close to the behavior of `typing.Self`, but there are some subtle downsides, depending on the type-checker)
4. keep it as-is (the quotes are indeed required), which is fine if the class is `@final`, but is suboptimal when the method is called from a subtype
5. remove the annotations, and either rely on the inference capabilities of type-checkers, or go for one of those classic [`pip install scipy-stubs`](https://github.com/jorenham/scipy-stubs) so you can tell your friends about how much time you just saved because of it ;)

Reply:
> keep it as-is (the quotes are indeed required), which is fine if the class is @final, but is suboptimal when the method is called from a subtype

Let's leave the quotes in for this PR, we may as well stick with sub optimality until scipy-stubs is integrated

Reply:
Then I think it's better to remove it in this case. Otherwise it could give users the idea that `scipy` is actually typed, and they'll miss out on `scipy-stubs`.